src/Connection/Replication/MasterSlaveReplication.php

```diff
@@ -72,7 +72,7 @@ class MasterSlaveReplication implements ReplicationInterface
     /**
      * {@inheritdoc}
      */
-    public function __construct(ReplicationStrategy $strategy = null)
+    public function __construct(?ReplicationStrategy $strategy = null)
```

Comment:
An idea: put these changes not strictly related to hash field expiration in a separate PR, to avoid noise?

Reply:
There's already a PR with this changes, I'm waiting for approval and then rebase this branch